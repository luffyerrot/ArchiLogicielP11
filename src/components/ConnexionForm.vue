<template>
    <h1 class="message"> {{ props.register ? 'Enregistrement' : 'Connexion' }} </h1>
    <v-card class="mx-auto px-6 py-8 mt-4" max-width="344">
      <v-form
        v-model="form"
        @submit.prevent="props.register ? onRegister() : onLogin()"
      >
        <v-text-field
          v-model="mail"
          :readonly="loading"
          :rules="[required]"
          class="mb-2"
          label="E-mail"
          type="email"
          clearable
        />

        <v-text-field
          v-model="password"
          :readonly="loading"
          :rules="[required]"
          label="Mot de passe"
          type="password"
          placeholder="Entrer le mot de passe"
          clearable
        />

        <br>

        <v-btn
          :disabled="!form"
          :loading="loading"
          color="success"
          size="large"
          type="submit"
          variant="elevated"
          block
        >
          {{ props.register ? "S'enregistrer" : "Se connecter" }}
        </v-btn>
      </v-form>
      <div v-if="msgConnexionError !== null">
        <p
          class="message mt-4"
          style="color: red"
        >
          {{ msgConnexionError }}
        </p>
      </div>
    </v-card>
</template>
  
<script setup lang="ts">
    import router from '@/router';
import { login, subscribe } from '@/services/AuthService';
    import { useUserStore } from '@/stores/user';

    const props = defineProps<{
        register: boolean
    }>()

    const user = useUserStore()

    const msgConnexionError: Ref<string | null> = ref(null);
    const form: Ref<boolean> = ref(false);
    const mail: Ref<string | null> = ref(null);
    const password: Ref<string | null> = ref(null);
    const loading: Ref<boolean> = ref(false);

    const connexionDto = computed(() => ({ mail: mail.value ? mail.value : '', motdepasse: password.value ? password.value : '' }));

    function onRegister() {
        loading.value = true;
        subscribe(connexionDto.value)
        .then(() => {
            msgConnexionError.value = null;
            router.push('/login');
        })
        .catch(() => msgConnexionError.value = 'Erreur lors de l\'inscription')
        .finally(() => loading.value = false);
    };
    
    function onLogin() {
        loading.value = true;
        login(connexionDto.value)
        .then(response => {
            user.updateUser(response.data);
            router.push('/mapfinder');
        })
        .catch(() => msgConnexionError.value = 'Erreur lors de la connexion')
        .finally(() => loading.value = false);
    };

    const required = (value: string) => {
        return !!value || 'Le champ est requis'
    };
</script>

<style scoped lang="scss">
  .message {
    text-align: center;
  }
</style>