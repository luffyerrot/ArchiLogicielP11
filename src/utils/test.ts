import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}