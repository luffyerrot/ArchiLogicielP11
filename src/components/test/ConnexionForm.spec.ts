import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import ConnexionForm from "../ConnexionForm.vue";

describe('Formulaire de connexion / enregistrement', () => {
    const wrapper = factory(ConnexionForm, true, true);
    renderCorrectly(wrapper);
})