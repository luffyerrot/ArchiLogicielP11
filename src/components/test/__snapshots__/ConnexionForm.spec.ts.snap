// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Formulaire de connexion / enregistrement > Bon affichage de la page 1`] = `
"<h1 data-v-e95e7a8d="" class="message">Connexion</h1>
<v-card-stub data-v-e95e7a8d="" class="mx-auto px-6 py-8 mt-4" maxwidth="344" disabled="false" flat="false" hover="false" ripple="true" border="false" density="default" loading="false" tile="false" replace="false" exact="false" tag="div" variant="elevated"></v-card-stub>"
`;
